src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -266,7 +287,11 @@ public void commitJob(JobContext jobContext) throws IOException {
                       writeDisposition, sourceUris, partitionType, timePartitioningType, range, partitionByField,
                       requirePartitionFilter, clusteringOrderList, tableExists, jobLabelKeyValue, conf);
       } catch (Exception e) {
-        throw new IOException("Failed to import GCS into BigQuery. ", e);
+        String errorMessage = "Failed to import GCS into BigQuery.";
+        throw ErrorUtils.getProgramFailureException(
+          new ErrorCategory(ErrorCategory.ErrorCategoryEnum.PLUGIN), errorMessage,
+          String.format(errorMessageFormat, ErrorPhase.COMMITTING, errorMessage), ErrorType.SYSTEM, true,
+          new IOException(errorMessage, e));
```

Comment:
There is no need to wrap actual exception in Ioexception here now.

Reply:
Similar comment in all places.

Reply:
Removed! 
I have made changes in a new commit to help with the review, i will squash those before merging.